### PR TITLE
Fixes #37112 - Remove query-string dependency

### PR DESCRIPTION
**Pull request by jeremylenz**
#### What are the changes introduced in this pull request?

- Remove the Katello-only dependency `query-string`
- Refactor Content > `ModuleStreamsPage` to a Hooks-based component
- remove no-longer-used tests
- remove no-longer-used React hooks

#### What are the testing steps for this pull request?

make sure everything still works

### CODE REVIEW

**Comment by jeremylenz** [comment]
oh I think tests are gonna fail now until https://github.com/Katello/katello/pull/10962 is in

**Comment by jeremylenz** [comment]
rebased

File: webpack/components/Content/ContentTable.js [hunk 1]
```diff
@@ -20,7 +20,6 @@ const ContentTable = ({ content, tableSchema, onPaginationChange }) => {
       loadingText={__('Loading')}
     >
       <Table
-        ouiaId="content-table-table"
```
Comment (chris1984) [inline webpack/components/Content/ContentTable.js:23]: Are these `ouiaId` part of `query-string`? I thought we needed them for QE to be happy?

**Review by chris1984 — COMMENTED** [review]
One small question, easier to read. Will start testing.

File: webpack/components/Content/ContentTable.js (hunk 1 above)
Comment (jeremylenz) [reply webpack/components/Content/ContentTable.js:23]: Normally we do need them, but this is a Patternfly 3 component and it causes a console error if I include it.

**Review by chris1984 — COMMENTED** [review]
I am still seeing the ouiaid error after applying the PR:

```javascript
Warning: React does not recognize the `ouiaId` prop on a DOM element. If you intentionally want it to appear in the DOM as a custom attribute, spell it as lowercase `ouiaid` instead. If you accidentally passed it from a parent component, remove it from the DOM element.
    in table (created by Provider)
    in Provider (created by TablePfProvider)
    in TablePfProvider (created by Table)
    in div (created by Table)
    in Table (created by ContentTable)
    in LoadingState (created by ContentTable)
    in ContentTable (created by GenericContentPage)
    in div (created by Col)
    in Col (created by GenericContentPage)
    in div (created by Row)
    in Row (created by GenericContentPage)
    in div (created by Grid)
    in Grid (created by GenericContentPage)
    in GenericContentPage (created by ModuleStreamsPage)
    in ModuleStreamsPage (created by Context.Consumer)
    in withRouter(ModuleStreamsPage) (created by ConnectFunction)
    in ConnectFunction (created by Headers)
    in Headers (created by CheckOrg)
    in CheckOrg (created by ConnectFunction)
    in ConnectFunction (created by Context.Consumer)
    in Route (created by _default)
    in div (created by _default)
    in _default (created by Application)
    in Router (created by BrowserRouter)
    in BrowserRouter (created by Application)
    in Application (created by ConnectFunction)
    in ConnectFunction (created by I18nProviderWrapper(Connect(Application)))
    in IntlProvider (created by I18nProviderWrapper(Connect(Application)))
    in I18nProviderWrapper(Connect(Application)) (created by StoreProvider(I18nProviderWrapper(Connect(Application))))
    in Provider (created by StoreProvider(I18nProviderWrapper(Connect(Application))))
    in StoreProvider(I18nProviderWrapper(Connect(Application))) (created by DataProvider(StoreProvider(I18nProviderWrapper(Connect(Application)))))
    in DataProvider(StoreProvider(I18nProviderWrapper(Connect(Application))))
    in AwaitedMount
```

**Comment by chris1984** [comment]
Other than that it works fine, module stream page works normally

**Review by chris1984 — APPROVED** [review]
ACK, works great and looks fine.

**Pull request merged**